sqlg-core/src/main/java/org/umlg/sqlg/structure/SqlgDataSourceFactory.java

```diff
@@ -0,0 +1,20 @@
+package org.umlg.sqlg.structure;
+
+import org.apache.commons.configuration.Configuration;
+import javax.sql.DataSource;
+
+/**
+ * Created by petercipov on 27/02/2017.
+ */
+public interface SqlgDataSourceFactory {
+
+    SqlgDataSource setup(String driver, final Configuration configuration) throws Exception;
+    SqlgDataSource setupFromJndi(String jndiName, Configuration configuration) throws Exception;
```

Comment:
OK, make sense. But original code did not handle it either.... 

From my point of view I have 2 options
1) Distiguish jndi connection by type and on all places datasource is used I will have to use if
2) leave as it is and let SqlgDataSource#get(String) encapsulate jndi / pooled datasources.

I am open to another ideas.